### DISCUSSION TITLE
Fan category

### DISCUSSION THREAD

**Discussion by Suitec1** [HomeSpan/HomeSpan#825] [Q&A]
Hello friends, I'm starting with this library and I've already managed to do something, but my goal looks like this: 1 LED PWM, 2 RGB and the third I would like a relay with which I would switch the fan,
I know I have it in the light category, but I don't know what to change it to so that in the application it is under the fan category and not the light.

Thank you very much.

`#include "HomeSpan.h" 
#include "DEV_LED.h"     

void setup() {

  // Example 10 illustrates how to control an RGB LED to set any color and brightness.
  // The configuration below should look familiar by now.  We've created a new derived Service,
  // called DEV_RgbLED to house all the required logic.  You'll find all the code in DEV_LED.h.
  // For completeness, this configuration also contains an on/off LED and a dimmable LED as shown
  // in prior examples.
  
  Serial.begin(115200);

  homeSpan.begin(Category::Bridges,"HomeSpan Bridge");

  new SpanAccessory();  
    new Service::AccessoryInformation();
      new Characteristic::Identify();

  new SpanAccessory();                                                          
    new Service::AccessoryInformation();
      new Characteristic::Identify();            
      new Characteristic::Name("Rele");
    new DEV_LED(16);                                                               // Create an On/Off LED attached to pin 16

  new SpanAccessory();                                                          
    new Service::AccessoryInformation();    
      new Characteristic::Identify();               
      new Characteristic::Name("LED"); 
    new DEV_DimmableLED(21);                                                       // Create a Dimmable (PWM-driven) LED using attached to pin 17

  new SpanAccessory();                                                          
    new Service::AccessoryInformation();    
      new Characteristic::Identify();               
      new Characteristic::Name("RGB LED"); 
    new DEV_RgbLED(32,22,23);                                                      // Create an RGB LED attached to pins 32,22,23 (for R, G, and B LED anodes)
      
} // end of setup()

//////////////////////////////////////

void loop(){
  
  homeSpan.poll();
  
} // end of loop()`

**Comment by HomeSpan** [comment]
For the Fan you should use a Fan Service as this will show the proper icon on the Accessory tile.  See the tutorial examples for details.

**Reply by Suitec1** [reply]
I looked at that, but I'm really only concerned with switching on the fan using a relay, is it not possible to have lights and a fan in one program?  I also tried the GPT chat but it's completely useless.  :(

**Comment by HomeSpan** [comment]
(Chat GPT won't be of any help for something like this)

What do you mean by "one program".  All three of your Accessories are already in the same program.  In your script above I assume the first functional Accessory labeled "Rele" is what you want to use for the Fan?  If so, simply a Fan Service instead of the LightBulb Service for that Accessory.  Note DEV_LED is based on the LightBulb Service.  If you to through the tutorials you will see analogous structures designed for a Fan.

**Reply by Suitec1** [reply]
Ok, I'll give it a good look tomorrow. :)
